python/lsst/analysis/tools/actions/plot/gridPlot.py

```diff
@@ -0,0 +1,133 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("GridPlot", "GridPanelConfig")
+
+from typing import TYPE_CHECKING
+
+import matplotlib.pyplot as plt
+from lsst.pex.config import Config, ConfigDictField, DictField, Field, ListField
+from lsst.pex.config.configurableActions import ConfigurableActionField
+from matplotlib.gridspec import GridSpec
+
+from ...interfaces import PlotAction, PlotElement
+
+if TYPE_CHECKING:
+    from lsst.analysis.tools.interfaces import KeyedData, PlotResultType
+
+
+class GridPanelConfig(Config):
+    plotElement = ConfigurableActionField[PlotElement](
+        doc="Plot element.",
+    )
+    title = DictField[str, str](
+        doc="String arguments passed into ax.set_title() defining the plot element title.",
+    )
+    titleY = Field[float](
+        doc="Y position of plot element title.",
+        default=None,
+    )
+
+
+class GridPlot(PlotAction):
+    """Plot a series of plot elements onto a regularly spaced grid."""
+
+    plotElements = ConfigDictField(
+        doc="Plot elements.",
+        keytype=int,
+        itemtype=GridPanelConfig,
+    )
+    numRows = Field[int](
+        doc="Number of rows.",
+        default=1,
+    )
+    numCols = Field[int](
+        doc="Number of columns.",
+        default=1,
+    )
+    xDataKeys = DictField[int, str](
+        doc="Dependent data definitions. The key of this dict is the panel ID. The values are keys of data "
+        "to plot (comma-separated for multiple) where each key may be a subset of a full key.",
+        default={},
+    )
+    valsGroupBy = DictField[int, str](
+        doc="Independent data definitions. The key of this dict is the panel ID. The values are keys of data "
+        "to plot (comma-separated for multiple) where each key may be a subset of a full key.",
+    )
+    figsize = ListField[float](
+        doc="Figure size.",
+        default=[8, 8],
+    )
+    dpi = Field[float](
+        doc="Dots per inch.",
+        default=150,
+    )
+    suptitle = DictField[str, str](
+        doc="String arguments passed into fig.suptitle() defining the figure title.",
+        optional=True,
+    )
+
+    def __call__(self, data: KeyedData, **kwargs) -> PlotResultType:
+        """Plot data."""
+        fig = plt.figure(figsize=self.figsize, dpi=self.dpi)
+        if self.suptitle is not None:
+            fig.suptitle(**self.suptitle)
+        gs = GridSpec(self.numRows, self.numCols, figure=fig)
+
+        for row in range(self.numRows):
+            for col in range(self.numCols):
+                index = row * self.numCols + col
+                if index not in self.valsGroupBy.keys():
+                    continue
+                ax = fig.add_subplot(gs[row, col])
+
+                xList = x.split(",") if (x := self.xDataKeys.get(index)) else None
+                valList = self.valsGroupBy[index].split(",")
```

Comment:
I think this anticipated one of my use cases.  If I specify `valsGroupBy = {0: "C10,C11,C12,C13", 1: "C00,C01,C02,C03"}`, this will fill two subplots with four amplifiers worth of data, correct?

Reply:
Indeed you can. Here is an example script that should "just work" for you in a notebook:

```
import lsst.daf.butler as dB
from lsst.analysis.tools.atools.cpVerifyQuantityProfile import (
    CPVerifyQuantityAmpProfileScatterTool,
    CPVerifyQuantityAmpProfileHistTool
)
butler = dB.Butler("/sdf/data/rubin/user/czw/ci_cpp_gen3/DATA")
biasT = butler.get("verifyBiasResults", instrument='LATISS', collections="u/kannawad/ci_cpv_bias")
atool = CPVerifyQuantityAmpProfileScatterTool()
atool.produce.plot.valsGroupBy = {0: "C00,C01,C02,C03", 5: "C10,C11,C12,C13"}
atool.prep.quantityKey = "biasSerialProfile"
atool.finalize()
atool(biasT)
```

As shown, I've modified the first cell to include data from `C00,C01,C02,C03`, and the fifth cell to include data from `C10,C11,C12,C13`. This produces a figure which looks like this:

![czwTest](https://github.com/lsst/analysis_tools/assets/22297403/8d8c6691-b790-43b8-821b-fe4fa9c7e824)

Reply:
(Note how the other cells are not drawn, so the `tight_layout` method applied at the end of `gridPlot` has shaved off the superfluous space).